src/download.py

```diff
@@ -1,27 +1,39 @@
 import asyncio
 import io
+import functools
 import os
 import random
-import time
+import shutil
 import sys
+import time
+import typing as t
+from concurrent.futures import ThreadPoolExecutor
 import aiohttp
 from yarl import URL
+from .utils import check_files_exist
 
 # some important tricks:
 # 1. os.sched_getaffinity to work right in docker
 # 2. memoryview for less copies
 # 3. keep redirects from the first head
+MIN_CHUNK_SIZE = 1024 * 32  # 32kb
```

Comment:
I'd likely make this much much bigger, like 1Mb or.... 10Mb, it's almost not worth considering a split lower than this regardless as a high concurrency for 32kb is not providing any real benefit.

Is this configurable from at least os.env? with our cache system in CW, it makes a lot of sense to set the min chunk_size ~50MB as that will optimize for the DiretIO floor we've set (we could lower the floor, but that gets into other Async related things), and for larger file the sweet spot is much larger [in the order of 100-250MB).